### PR TITLE
Add FW filter support

### PR DESCRIPTION
**Pull request by chantra**
Add support for FW filter.

### CODE REVIEW

**Comment by chantra** [comment]
I will add documentation at a later stage but on this PR. Still, I created the PR so feedback can be given in the meantime.

File: filter.go [hunk 1]
```diff
@@ -39,6 +41,89 @@ func (filter *U32) Type() string {
...
+	InDev     string
+	Mask      uint32
+	Index     uint32
+	Buffer    uint32
+	Mtu       uint32
+	Mpu       uint16
+	Rate      uint32
+	AvRate    uint32
+	PeakRate  uint32
+	Action    int32
```
Comment (vishvananda) [inline filter.go]: generally i try to keep the signed ints at this layer as "int" instead of int32 but that is minor

File: filter_linux.go [hunk 2]
```diff
@@ -1,10 +1,11 @@
 package netlink
 
 import (
+	"bytes"
+	"encoding/binary"
 	"fmt"
-	"syscall"
-
 	"github.com/vishvananda/netlink/nl"
+	"syscall"
```
Comment (vishvananda) [inline filter_linux.go]: this is out of order

**Comment by vishvananda** [comment]
looks good overally. Some minor stuff inline

**Pull request merged**